underdocs-version/src/main/kotlin/underdocs/version/retriever/VersionRetriever.kt

```diff
@@ -0,0 +1,11 @@
+package underdocs.version.retriever
+
+import underdocs.version.Version
+
+interface VersionRetriever {
+    companion object {
+        fun create(): VersionRetriever = DefaultVersionRetriever()
+    }
```

Comment:
Nice use of companion objects and the factory pattern! 👍

Reply:
Just followed the instructions provided.